azurerm/internal/services/resource/parse/group.go

```diff
@@ -0,0 +1,45 @@
+package parse
+
+import (
+	"fmt"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+)
+
+type ResourceGroupId struct {
+	SubscriptionId string
+	Name           string
+}
+
+func (id ResourceGroupId) ID(_ string) string {
+	fmtString := "/subscriptions/%s/resourceGroups/%s"
+	return fmt.Sprintf(fmtString, id.SubscriptionId, id.Name)
+}
+
+func NewResourceGroupID(subscriptionId, name string) ResourceGroupId {
+	return ResourceGroupId{
+		SubscriptionId: subscriptionId,
+		Name:           name,
+	}
+}
+
+func ResourceGroupID(input string) (*ResourceGroupId, error) {
+	id, err := azure.ParseAzureResourceID(input)
+	if err != nil {
+		return nil, fmt.Errorf("[ERROR] Unable to parse Resource Group ID %q: %+v", input, err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("unable to parse Resource Group ID %q: %+v", input, err)
```

Reply:
this'll get replaced with a generated version shortly, so I'm going to suggest we wait until then for that